### ISSUE TITLE
importGlobalAveragePool ERROR

### ISSUE THREAD

**Issue by jianyin2016** [NVIDIA/TensorRT#402]
parse onnx model failed
=================
I believe my onnx model is exported without dynamic shape by setting `dynamic_axes=None`
-------------------------------------------------------------------------------------------------------
*pytorch to onnx export api are called as follows:*

    torch.onnx.export(model, inputs, './segment.onnx',
                              export_params=True,
                              verbose=True,
                              training=False,
                              input_names=None,
                              output_names=None,
                              operator_export_type=torch.onnx.OperatorExportTypes.ONNX,
                              opset_version=11,
                              do_constant_folding=True,
                              example_outputs=torch_out,
                              strip_doc_string=True,
                              dynamic_axes=None,
                              keep_initializers_as_inputs=True
                              )

*but when parse onnx model with TensorRt api, function `createNetwork()` and `createNetworkV2(0U)` ends up with the same error:*

---
>ERROR: ModelImporter.cpp:457 In function importModel:
[4] Assertion failed: !_importer_ctx.network()->hasImplicitBatchDimension() && "This version of the
ONNX parser only supports TensorRT INetworkDefinitions with an explicit batch dimension. Please ensure the network was created using the EXPLICIT_BATCH NetworkDefinitionCreationFlag."

*changing the `createNetwork()` function to `createNetworkV2()` by setting the explict_batch flag like*

    INetworkDefinition* network = builder->createNetworkV2(1U <<
		static_cast<int>(NetworkDefinitionCreationFlag::kEXPLICIT_BATCH));
*this results in another error:*

>While parsing node number 645 [GlobalAveragePool]:
ERROR: builtin_op_importers.cpp:1087 In function importGlobalAveragePool:
[8] Assertion failed: !isDynamic(kernelSize) && "Cannot run global average pool on an input with dynamic spatial dimensions!" 

*onnx model verbose shows no GlobalAveragePlooing dynamic shape sign*

>%1258 : Float(1, 128, 1, 1) = onnx::GlobalAveragePool

detailed enviroment information
--------------------------------------

---
>OS : Windows 7
TensorRt version : 7.0.11
pyTorch version : 1.4.0
cuda : 10.2
ONNX IR version : 0.0.4
Opset version : 11
Producer name :  pytorch
Producer version : 1.3

*BTW,using trtexec.exe araise similar error* 

>While parsing node number 645 [GlobalAveragePool]:
ERROR: builtin_op_importers.cpp:1087 In function importGlobalAveragePool:
[8] Assertion failed: !isDynamic(kernelSize) && "Cannot run global average pool on an input with dynamic spatial dimensions!"
[02/28/2020-16:28:50] [E] Failed to parse onnx file
[02/28/2020-16:28:50] [E] Parsing model failed
[02/28/2020-16:28:50] [E] Engine creation failed
[02/28/2020-16:28:50] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec # E:\TensorRT-7.0.0.11\samples\trtexec\\..\..\bin\trtexec.exe --onnx=E:\models\segment.onnx

**Comment by rmccorm4** [comment]
Hi @jianyin2016 ,

I'm not sure on the specifics since I don't have the model, but it might be possible that an intermediate layer has shape computed by the previous layer's output (dynamic shape), even if you didn't specify the input layer to have a dynamic axis.

However, even if is this case, I believe this has been fixed in TensorRT OSS in this commit: https://github.com/stevenlix/onnx-tensorrt/commit/1dc27c14043536daa2239f4f38f4102f380c7d35

You can try to build the OSS components on top of your TensorRT release. It would look something like this in a container for TensorRT 7 on Linux:

```
# Start TensorRT 7 container
nvidia-docker run -it -v $PWD:/mnt --workdir=/mnt nvcr.io/nvidia/tensorrt:20.01-py3

# Build OSS Components from https://github.com/NVIDIA/TensorRT
wget https://raw.githubusercontent.com/rmccorm4/tensorrt-utils/master/OSS/build_OSS.sh
source build_OSS.sh

# Parse model
trtexec --explicitBatch --onnx=segment.onnx
```

I'm not sure if it's possible to build OSS components on Windows, but please share if you're able to do so 🙂

**Comment by jianyin2016** [comment]
Hi,@rmccorm4 ,
I  believe that this error isn't origined from the network components, cause the previsous laysers are mainly ConvBnReLu modules. I will be pleased to construct a minimal netowrk to reproduce the error if needed.

Besides, I have noted the commit that may solve my question. but As far as I know,  for Windows , there is few build tutorial available, build OSS on Windows is thus supposed to be a time-consuming ，boring work and possiblly ends up with a failure. So I choose to wait for the next minor update for Windows users, Is there any schedule? :)

Thanks for your time.
Nice weekend.

**Comment by charleybian** [comment]
Same problem on TensorRT-7.1.3.4, while old 7.0.0.11 is fine.
Debian 9
Try to upgrade to 7.1, and come across the problem:

[08/20/2020-15:19:11] [W] [TRT] /trt/TensorRT/parsers/onnx/onnx2trt_utils.cpp:216: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
While parsing node number 12 [GlobalAveragePool]:
ERROR: /trt/TensorRT/parsers/onnx/builtin_op_importers.cpp:1086 In function importGlobalAveragePool:
[8] Assertion failed: !isDynamic(kernelSize) && "Cannot run global average pool on an input with dynamic spatial dimensions!"
terminate called after throwing an instance of 'std::runtime_error'
  what():  TRT_API: Failed to parse the ONNX model file

Could anyone help?

**Comment by rajeevsrao** [comment]
@jianyin2016 @charleybian the issue you described was likely fixed with [this commit](https://github.com/onnx/onnx-tensorrt/commit/1dc27c14043536daa2239f4f38f4102f380c7d35#diff-b23eabdc4ee3ebabdbb607535b2ed80c471093b0fef41a5b53bce876ca9ec96d).  Can you please conform, and if not please provide a repro test?

**Comment by jianyin2016** [comment]
Hi, @rajeevsrao ,

Yeah, I think the fix in TRT7.2 works, as reported in [resnet-onnx-parse-error](https://github.com/NVIDIA/TensorRT/issues/658#issue-650309109) .
